ovs/lib/storagerouter.py

```diff
@@ -632,7 +587,54 @@ def add_vpool(parameters):
             vpool.save()
             raise ValueError('Something went wrong trying to calculate the fragment cache size')
 
+        root_client = ip_client_map[storagerouter.ip]['root']
         root_client.dir_create(dirs2create)
+        partitions_lock.release()
+
+        #################
+        # ARAKOON SETUP #
+        #################
+        counter = 0
+        while counter < 300:
+            try:
+                if StorageDriverController.manual_voldrv_arakoon_checkup() is True:
+                    break
+            except Exception:
+                StorageRouterController._logger.exception('Arakoon checkup for voldrv cluster failed')
+                vpool.status = VPool.STATUSES.FAILURE
+                vpool.save()
+                raise
+            counter += 1
+            time.sleep(1)
+            if counter == 300:
+                vpool.status = VPool.STATUSES.FAILURE
+                vpool.save()
+                raise RuntimeError('Arakoon checkup for the StorageDriver cluster could not be started')
+
+        ####################
+        # CLUSTER REGISTRY #
+        ####################
+        node_configs = []
+        existing_storagedrivers = []
+        for sd in vpool.storagedrivers:
+            if sd != storagedriver:
+                existing_storagedrivers.append(sd)
+            sd.invalidate_dynamics('cluster_node_config')
+            node_configs.append(ClusterNodeConfig(**sd.cluster_node_config))
+
+        try:
+            vpool.clusterregistry_client.set_node_configs(node_configs)
+            for sd in existing_storagedrivers:
+                vpool.storagedriver_client.update_cluster_node_configs(str(sd.storagedriver_id), req_timeout_secs=10)
+        except:
+            for sd_partition in storagedriver.partitions:
+                sd_partition.delete()
+            storagedriver.delete()
```

Comment:
Can you check whether this is the right thing to do? Maybe we can be smarter and cleanup the mess to we have a RUNNING (or none) vPool left.